packages/starlight/integrations/asides.ts

```diff
@@ -5,6 +5,21 @@ import remarkDirective from 'remark-directive';
 import type { Plugin, Transformer } from 'unified';
 import { remove } from 'unist-util-remove';
 import { visit } from 'unist-util-visit';
+import type { StarlightConfig } from '../types';
+
+/** get current lang from file full path */
+function getLocaleFromPath(path: string, config: StarlightConfig): string {
+  // format path to unix style path
+  const formatPath = path.replace(/\\/g, '/');
+	const parts = formatPath.split('/');
+	const langIndex = parts.findIndex((part) => /^[a-z]{2}(-[a-z]{2})?$/.test(part));
+	if (langIndex !== -1) {
+		return parts[langIndex] || 'en';
+	} else {
+		const defaultLocale = config.defaultLocale?.lang || config.defaultLocale?.locale;
+		return defaultLocale || 'en';
+	}
+}
```

Comment:
There are some differences here from our existing [slug utilities](https://github.com/withastro/starlight/blob/0100298b158f90cf2beed67e9a185f40a533465b/packages/starlight/utils/slugs.ts). Importantly, we support locales that don’t match the BCP `aa-AA` shape and also because `path` here is absolute (at least in my testing), we’d risk clashing with some other directory in the path.

I think the right approach is to use `root` and `srcDir` from the Astro config. With those we can strip the path down so we only get the path inside `content/docs/` and then apply the same logic we do for file IDs.

Something like this:

```js
function pathToLocale(slug: string, config: StarlightConfig): string | undefined {
	const locales = Object.keys(config.locales || {});
	const baseSegment = slug.split('/')[0];
	if (baseSegment && locales.includes(baseSegment)) return baseSegment;
	return undefined;
}

function localeToLang(locale: string | undefined, config: StarlightConfig): string {
	const lang = locale ? config.locales?.[locale]?.lang : config.locales?.root?.lang;
	const defaultLang = config.defaultLocale?.lang || config.defaultLocale?.locale;
	return lang || defaultLang || 'en';
}

/** get current lang from file full path */
function getLocaleFromPath(
	unformattedPath: string,
	starlightConfig: StarlightConfig,
	astroConfig: AstroConfig
): string {
	const srcDir = new URL(astroConfig.srcDir, astroConfig.root);
	const docsDir = new URL('content/docs/', srcDir);
	const path = unformattedPath
		// Format path to unix style path.
		.replace(/\\/g, '/')
		// Strip docs path leaving only content collection file ID.
		// Example: /Users/houston/repo/src/content/docs/en/guide.md => en/guide.md
		.replace(docsDir.pathname, '');
	const locale = pathToLocale(path, starlightConfig);
	return localeToLang(locale, starlightConfig);
}
```

To do that we’ll need to pass the Astro config object through from `index.ts` too.

Reply:
Yeah. Maybe we don't need `localeToLang` because it's handled in `useTranslations`?

Reply:
Good point! Yeah that just needs the locale I think.